### DISCUSSION TITLE
How to show privacy policy page linked from the footer when disablePageInteraction is true?

### DISCUSSION THREAD

**Discussion by akosphp** [orestbida/cookieconsent#658] [Q&A] [answered]
Hi all, 

I included a link to our privacy policy page in the consent modal's footer. The disablePageInteraction parameter is set to true, leading to some interesting situations. The privacy policy is a regular page on the site, so when the user clicks the link on the modal, the privacy policy page loads, but they cannot read it before accepting the modal. I understand that this is how the disablePageInteraction parameter functions. We want to maintain this strict mode, but we need to find a way for users to view the privacy policy before accepting anything. 

What is the best way to resolve this situation?

**Comment by orestbida** [comment, marked as answer]
Hi @akosphp,

you need to differentiate the privacy policy page from the other pages.

There are many ways to do this. One approach is to add a simple script tag in the privacy policy page only, and set a unique global variable:

```html
<script>window.isPrivacyPage = true;</script>
```

The above script must be placed before the plugin's initialization script. 
You can then toggle the `disablePageInteraction` option dynamically like so:

```js
CookieConsent.run({
    /// ...
    disablePageInteraction: !window.isPrivacyPage
});
```

**Reply by akosphp** [reply]
Thank you very much for the response and the time you've put into it!

**Comment by jsiemon-edv** [comment]
Gentlemen,

as far as I understand current consensus in rulings, at least in Germany, pages like "Imprint" and "Privacy Policy" must be accessible and readable without obstruction by any kind before any consent is given.
The solution above works fantastic, of course, if your cookie modal comes as a box, confined to an edge or corner of the window on large screen devices. On smaller screens, however, when using a layout option with positioning the modal center screen or otherwise modifying it, it may become a problem.

Therefore I would like to suggest an addendum to the authors solution. Prevent the modal to pop up on these pages entirely by setting the option "autoShow" as well:
 
```
CookieConsent.run({
    /// ...
    autoShow: !window.isPrivacyPage,
    disablePageInteraction: !window.isPrivacyPage
    /// ...
});
```

**Reply by osworx** [reply]
Not only in Germany, the GDPR (DSGVO) tells you that (also).
Pages like Imprint and Contact **MUST** be reach- and readable within (at least) one click any time!

That means, hiding the consent modal (or any similiar or any other popup) on such pages is required to fullfill current (and coming) Laws.

Not to forget @orestbida : not every user can edit pages or/and add code like you suggesting.
Means, CookieConsent must provide an easy way for that.
